src/theme/components/checkbox.ts

```diff
@@ -121,44 +135,45 @@ const baseStyleControl = {
       color: "dark.ui.error.primary",
     },
   },
-};
-
-// Style object for the Checkbox's label
-const baseStyleLabel = checkboxRadioLabelStyles;
-
-// Style object for the Checkbox's helper text
-const baseStyleHelperErrorText = checkboxRadioHelperErrorTextStyle;
-
-const baseStyle = {
+});
+
+// // Style object for the Checkbox's label
+const baseStyleLabel = defineStyle({
+  cursor: "pointer",
+  verticalAlign: "top",
+  ...checkboxRadioLabelStyles,
+});
+
+const baseStyle = definePartsStyle({
+  // Style object for the Checkbox's helper text
+  helperErrorText: checkboxRadioHelperErrorTextStyle,
   icon: baseStyleIcon,
+  container: checkboxRadioHoverStyles,
   control: baseStyleControl,
   label: baseStyleLabel,
-  helperErrorText: baseStyleHelperErrorText,
-  ...checkboxRadioHoverStyles,
-};
+});
 
 // Sticking to "lg" for the default size.
 const sizes = {
   lg: {
     // Controls the width/height of the checkbox itself.
-    control: {
+    control: defineStyle({
       ...checkboxRadioControlSize,
       borderRadius: "checkbox",
-    },
+    }),
     // Controls the font-size of the label only.
-    label: { fontSize: "desktop.label.label1" },
+    label: defineStyle({ fontSize: "desktop.label.label1" }),
   },
 };
 
-const Checkbox = {
-  parts: ["helperErrorText", "icon", "control", "label"],
+const Checkbox = defineMultiStyleConfig({
   baseStyle,
   sizes,
   // Default values
   defaultProps: {
     size: "lg",
     colorScheme: "white",
   },
-};
+});
 
 export default Checkbox;
```

Comment:
Why didn't this need to be defined as `CustomCheckbox` in the theme `index.tx`?

Reply:
I _thought_ it would need to but I don't understand how the Chakra component naming pattern works. It's not consistent where this can be `Checkbox`, but there are other components where it seems Chakra doesn't like the name to conflict. I generally try the original name and if it doesn't work, I rename it only in the theme `index.ts` file.